packages/shared/lib/hooks/hooks.ts

```diff
@@ -11,10 +11,40 @@ import integrationPostConnectionScript from '../integrations/scripts/connection/
 import webhookService from '../services/notification/webhook.service.js';
 import { SpanTypes } from '../utils/telemetry.js';
 import { isCloud, isLocal, isEnterprise } from '../utils/temp/environment/detection.js';
+import { getSyncConfigsWithConnectionsByEnvironmentIdAndProviderConfigKey } from '../services/sync/config/config.service.js';
 import type { Result } from '../utils/result.js';
 import { resultOk, resultErr } from '../utils/result.js';
 import { NangoError } from '../utils/error.js';
 
+export const connectionCreationStartCapCheck = async ({
+    providerConfigKey,
+    environmentId
+}: {
+    providerConfigKey: string | undefined;
+    environmentId: number;
+}): Promise<boolean> => {
+    if (!providerConfigKey) {
+        return false;
+    }
+
+    const scriptConfigs = await getSyncConfigsWithConnectionsByEnvironmentIdAndProviderConfigKey(providerConfigKey, environmentId);
```

Comment:
opinion: not a huge fan of overly descriptive function, we have the parameters so we already know that info. `getConnectionsCount()` is more useful imo and directly state its raison d'être